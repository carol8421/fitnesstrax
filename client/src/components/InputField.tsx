import React from "react"

export interface Props {
  value: string | null
  onChange: (_: { value: string }) => void
  classNames: Array<string>
  placeholder: string
}

const InputField: React.SFC<Props> = ({
  classNames,
  onChange,
  value,
  placeholder,
}: Props) => (
  <input
    type="text"
    value={value || undefined}
    className={classNames.join(" ")}
    onChange={ev => onChange({ value: ev.target.value })}
    placeholder={placeholder}
  />
)

export default InputField
