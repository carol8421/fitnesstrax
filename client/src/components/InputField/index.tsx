import { Option } from "ld-ambiguity"
import React, { ChangeEvent } from "react"

import * as csn from "../../classnames"

export interface Props {
  value: string
  onChange: (_: string) => void
  classNames?: csn.ClassNames
  placeholder: string
}

const InputField: React.SFC<Props> = ({
  classNames,
  onChange,
  value,
  placeholder,
}: Props) => (
  <input
    type="text"
    value={value}
    className={csn.classnames(classNames ? classNames : {})}
    onChange={(ev: ChangeEvent<HTMLInputElement>) => onChange(ev.target.value)}
    placeholder={placeholder}
  />
)

export default InputField
