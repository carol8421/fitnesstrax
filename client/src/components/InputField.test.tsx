import React from "react"
import { shallow } from "enzyme"

import Option from "../option"
import InputField from "./InputField"

describe("InputField", () => {
  it("renders", () => {
    const onChange = jest.fn()
    const wrapper = shallow(
      <InputField
        value={Option.Some("sample-token")}
        classNames={{}}
        onChange={onChange}
        placeholder="placeholder text"
      />,
    )
    expect(wrapper).toMatchSnapshot()
    expect(onChange).not.toHaveBeenCalled()
  })

  it("calls the change handler when an update happens", () => {
    const onChange = jest.fn()
    const wrapper = shallow(
      <InputField
        value={Option.Some("sample-token")}
        classNames={{}}
        onChange={onChange}
        placeholder="placeholder text"
      />,
    )
    wrapper.simulate("change", { target: { value: "abcdefg" } })
    expect(onChange).toHaveBeenCalledWith("abcdefg")
  })
})
