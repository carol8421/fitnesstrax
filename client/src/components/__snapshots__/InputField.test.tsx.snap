// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders 1`] = `
<input
  className=""
  onChange={[Function]}
  placeholder="placeholder text"
  type="text"
  value="sample-token"
/>
`;
